### DISCUSSION TITLE
Ignore frequency in favor of recently opened

### DISCUSSION THREAD

**Discussion by c0de-intention** [danielfalk/smart-open.nvim#91] [Q&A] [answered]
Hi. I would like to say first I love this plugin. I am just wondering if there is a way when I just open it just to see only recent opened in that order only. I had a look at the code and it looks like each thing has weights. It would be great if I had access to those in order to make frequency 0. Personally I find recency to be much more relevant to my workflow. I am happy to put up a pr if there is a preferred way to implement this.

**Comment by danielfalk** [comment, marked as answer]
No, there isn't necessarily a way to do that, but you could just use regular `:Telescope oldfiles` for that.  It sounds like exactly what you want.

However, smart-open is designed to adapt its weights as you use it.  If you open up the picker and select the most recent file over the most frequent one, then smart-open adjusts those weights accordingly.  After several times of selecting the recent file, the weight of frequency should drop to a negligible amount.  And yes, you can also just go into the db and set other weights to `0` if you want to accelerate this process.

But yeah I'd recommend using something purpose-built for a specific criterion if you know you care about only a specific thing.

**Reply by c0de-intention** [reply]
Thanks for the super quick reply. I like this plugin because I can get my recent files but also search any file in the same place. This was one thing that vscode does well. I think old files does not allow this if not mistaken

**Reply by danielfalk** [reply]
That's true.  Still, due to the adaptive nature of this plugin, assuming you opt for the most recent file always, it will adapt and those files will become the top items fairly quickly.  The key is to move down to the (recent, not frequent) file you want and select it even if it's not at the top.  The plugin notices that you did that and boosts the recency weight and lowers the frequency weight.
